subprojects/core/src/main/java/org/gradle/internal/classpath/DefaultCachedClasspathTransformer.java

```diff
@@ -184,19 +179,17 @@ public List<URL> transform(Collection<URL> urls, StandardTransform transform) {
             return ImmutableList.of();
         }
         ClasspathFileTransformer transformer = fileTransformerFor(transform);
-        InstrumentationTypeRegistry typeRegistry = typeRegistryFactory.createFor(urls, transformer);
```

Comment:
💭 This part of the code was a source of a regression. This code is from first implementation of property upgrades (without artifact transforms) and upgrading `ReplacedBy` methods triggered re-discovering of classes in all jars.

Another proof that we should delete old instrumentation logic, just to avoid such bugs.

Reply:
It did show in the graph, but since this code runs logic via managaed executor on a different thread it was a different smaller stack that looked unimportant at first: other parts of graph seemed more affected. 

I would guess that because it starts extra threads it takes resources from other due to that also other were shown slower.